### PR TITLE
[Merged by Bors] - Improve stability of system tests

### PR DESCRIPTION
**Pull request by fasmat**
## Motivation

This PR tries to improve unstable system tests

## Description

- `sendTransactions` should now cancel when something goes wrong much earlier. Before it would occasionally not catch the `stop` layer and continue indefinitely. Now it stopes on any layer on or after `stop`. Additionally the context used to query the nodes via GRPC is also cancelled in a timely fashion to prevent a test from getting stuck here.
- I tried to get rid of the hard coded sleep in `watchLayers` by changing the asserted status from status approved to  applied but even when a layer is applied an address can still not be spawned - so I re-added the sleep.
- Updated the images used during system tests: `old_smesher`, `post_init`, `post_service`, `certifier` since they were all outdated already
- Decreased the number of nodes used in a typical test from 30 to 20, reduced poets in tests from 3 to 2.
- Added additional logging information in the fetcher, because in some tests transactions that were seemingly fetched from peers are later not found in the DB and not fetched again (still investigating)
- Made test runner more verbose - instead of only printing logs from failing tests it now prints logs from all tests, to avoid investigating the wrong test for a failed test run (should be reverted when tests are more stable again)
- Set `GOMAXPROCS` from a hard coded 4 to `limits.cpu` set on the k8s cluster
- Fixed a possible deadlock in partition tests where a go channel can run full causing the test to never complete
- Persisting active sets in the proposal builder can cause `SQL_BUSY` errors during system tests so I changed it to `TxImmediate` for now
- If transactions fail to be sent (because e.g. they were sent to the node before the account they are referencing has been spawned) the node now stops waiting for the transaction results in a timely manner instead of until the timeout of the test hits
- Instead of sleeping between spawning an account and sending a transaction I changed the code to wait at least 2 layers and then send the first transaction.

## Test Plan

- system tests pass

## TODO

- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [x] Update documentation as needed
- [x] Update [changelog](../CHANGELOG.md) as needed

### CODE REVIEW

**Review by acud — APPROVED** [review]

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors try

**Comment by fasmat** [comment]
bors cancel

**Comment by fasmat** [comment]
bors try

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Review by poszu — APPROVED** [review]

**Comment by poszu** [comment]
You could also drop the number of poets from 3 to 2 (or even 1)

**Comment by fasmat** [comment]
bors try

**Comment by fasmat** [comment]
bors try

**Comment by fasmat** [comment]
bors try

**Comment by fasmat** [comment]
bors try

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Comment by poszu** [comment]
@fasmat how about deploying more nodes in the "remote" mode (post-service in a separate container, not supervised)? It should help if the fighting for the CPU of go-sm and post-svc running in the "supervised" causes issues.

**Comment by fasmat** [comment]
@poszu I already reduced the number of nodes that run supervised post and increase the number that run remote post - before 18 and 2, now 13 and 5. I'll make it about half/half and see if that changes things.

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors try

File: sql/layers/layers.go [hunk 1]
```diff
@@ -96,14 +103,18 @@ func GetLatestStateHash(db sql.Executor) (rst types.Hash32, err error) {
...
 	}
 	return rst, err
 }
 
 // GetStateHash loads state hash for the layer.
 func GetStateHash(db sql.Executor, lid types.LayerID) (rst types.Hash32, err error) {
-	if rows, err := db.Exec("select state_hash from layers where id = ?1;",
+	log.With().Info("get state hash",
+		log.Uint32("layer", lid.Uint32()),
+	)
```
Comment (fasmat) [inline sql/layers/layers.go:115]: Remove again if `bors try` passes

File: sql/layers/layers.go [hunk 2]
```diff
@@ -75,14 +77,19 @@ func UnsetAppliedFrom(db sql.Executor, lid types.LayerID) error {
...
 		func(stmt *sql.Statement) {
 			stmt.BindInt64(1, int64(lid))
 			stmt.BindBytes(2, hash[:])
 		}, nil); err != nil {
 		return fmt.Errorf("set applied %s: %w", lid, err)
 	}
+	log.With().Info("set state hash",
+		log.Uint32("layer", lid.Uint32()),
+		log.ShortStringer("hash", hash),
+	)
```
Comment (fasmat) [inline sql/layers/layers.go:92]: Remove again if `bors try` passes

**Comment by fasmat** [comment]
bors try

**Comment by fasmat** [comment]
bors merge

**Pull request closed without merging**